Submitted By: Tushar Teredesai <tushar@linuxfromscratch.org>
Date: 2005-06-11
Initial Package Version: 8.4.10
Origin: TCL CVS
Upstream Status: In CVS
Description: Fix compilation when --enable-threads option is specified.
See <http://aspn.activestate.com/ASPN/Mail/Message/tcl-core/2619395>.

diff -Naur tcl8.4.10.orig/unix/tclUnixNotfy.c tcl8.4.10/unix/tclUnixNotfy.c
--- tcl8.4.10.orig/unix/tclUnixNotfy.c	2005-06-01 18:12:21.000000000 +0000
+++ tcl8.4.10/unix/tclUnixNotfy.c	2005-06-11 17:00:31.470883728 +0000
@@ -275,7 +275,7 @@
      */
 
     if (notifierCount == 0) {
-	int result;
+	int result, ignored;
 	if (triggerPipe < 0) {
 	    panic("Tcl_FinalizeNotifier: notifier pipe not initialized");
 	}
@@ -294,7 +294,7 @@
 	close(triggerPipe);
 
 	Tcl_ConditionWait(&notifierCV, &notifierMutex, NULL);
-	result = Tcl_JoinThread(notifierThread);
+	result = Tcl_JoinThread(notifierThread, &ignored);
 	if (result) {
 	    Tcl_Panic("Tcl_FinalizeNotifier: unable to join notifier thread");
 	}
